src/main/java/org/folio/service/orders/flows/update/open/OpenCompositeOrderPieceService.java

```diff
@@ -161,36 +162,22 @@ public Future<Void> updatePieceRecord(Piece piece, RequestContext requestContext
    * @return CompletableFuture with void.
    */
   public Future<Void> openOrderUpdateInventory(CompositePoLine compPOL, Piece piece, boolean isInstanceMatchingDisabled, RequestContext requestContext) {
-    if (Boolean.TRUE.equals(compPOL.getIsPackage())) {
-      return titlesService.getTitleById(piece.getTitleId(), requestContext)
-        .compose(title -> titlesService.updateTitleWithInstance(title, isInstanceMatchingDisabled, requestContext).map(title::withInstanceId))
-        .compose(title -> {
-          if (piece.getHoldingId() != null) {
-            return Future.succeededFuture(piece.getHoldingId());
-          }
-          if (!PoLineCommonUtil.isHoldingsUpdateRequired(compPOL)) {
-            return Future.succeededFuture();
-          }
-          return inventoryHoldingManager.createHoldingAndReturnId(title.getInstanceId(), piece.getLocationId(), requestContext)
-            .map(holdingId -> {
-              piece.setLocationId(null);
-              piece.setHoldingId(holdingId);
-              return holdingId;
-            });
-        })
-        .compose(holdingId -> {
-          if (PoLineCommonUtil.isItemsUpdateRequired(compPOL)) {
-            return inventoryItemManager.openOrderCreateItemRecord(compPOL, holdingId, requestContext);
-          }
-          return Future.succeededFuture();
-        })
-        .onSuccess(itemId -> Optional.ofNullable(itemId).ifPresent(piece::withItemId))
-        .mapEmpty();
-    }
-    else
-    {
+    if (!Boolean.TRUE.equals(compPOL.getIsPackage())) {
       return inventoryItemManager.updateItemWithPieceFields(piece, requestContext);
     }
+    var locationContext = createContextWithNewTenantId(requestContext, piece.getReceivingTenantId());
+    return titlesService.getTitleById(piece.getTitleId(), requestContext)
+      .compose(title -> titlesService.updateTitleWithInstance(title, isInstanceMatchingDisabled, locationContext, requestContext).map(title::withInstanceId))
+      .compose(title -> piece.getHoldingId() != null || !PoLineCommonUtil.isHoldingsUpdateRequired(compPOL)
+        ? Future.succeededFuture(piece.getHoldingId())
+        : inventoryHoldingManager.createHoldingAndReturnId(title.getInstanceId(), piece.getLocationId(), locationContext)
+          .map(holdingId -> piece.withLocationId(null).withHoldingId(holdingId).getHoldingId()))
```

Comment:
a bit harder to read code now, it is necessary to apply more energy to understand it
But maybe it is just for me, resolving comment in any case because logically it is the same as before